src/main/java/uk/gov/digital/ho/hocs/application/RequestData.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.digital.ho.hocs.application;
+
+import org.slf4j.MDC;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Objects;
+import java.util.UUID;
+
+@Component
+public class RequestData implements HandlerInterceptor {
+
+    static final String CORRELATION_ID_HEADER = "X-Correlation-Id";
+
+    static final String USER_ID_HEADER = "X-Auth-UserId";
+
+    static final String USERNAME_HEADER = "X-Auth-Username";
+
+    static final String GROUP_HEADER = "X-Auth-Groups";
+
+    static final String USER_ROLES_HEADER = "X-Auth-Roles";
+
+    private static final String ANONYMOUS = "anonymous";
+
+    private static final String BLANK = "";
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
```

Comment:
This method is set as returning a boolean, but only ever returns `true`, does it need to be boolean or could it be void?

Reply:
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/servlet/HandlerInterceptor.html 

It's intentionally true on the interface so that you can fail a request if it doesn't satisfy requirements, for example with security.